commit f5bf1fdaf36163d5c2f1b9b51df96326ebbb0e9c
Author: Peter Hutterer <peter.hutterer@who-t.net>
Date:   Fri Feb 13 09:56:22 2009 +1000

    xkb: Fix wrong colour reference in XKB geometry copying. #20081
    
    base_color and label_color need to reference the color in the destination, not
    in the source.
    
    X.Org Bug 20081 <http://bugs.freedesktop.org/show_bug.cgi?id=20081>
    
    Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
    Signed-off-by: Daniel Stone <daniel@fooishbar.org>

diff --git a/xkb/xkbUtils.c b/xkb/xkbUtils.c
index aa63b35..9a1edc9 100644
--- a/xkb/xkbUtils.c
+++ b/xkb/xkbUtils.c
@@ -1955,9 +1955,9 @@ _XkbCopyGeom(XkbDescPtr src, XkbDescPtr dst)
 
             strcpy(dst->geom->label_font, src->geom->label_font);
             i = XkbGeomColorIndex(src->geom, src->geom->label_color);
-            dst->geom->label_color = &(src->geom->colors[i]);
+            dst->geom->label_color = &(dst->geom->colors[i]);
             i = XkbGeomColorIndex(src->geom, src->geom->base_color);
-            dst->geom->base_color = &(src->geom->colors[i]);
+            dst->geom->base_color = &(dst->geom->colors[i]);
         }
         else {
             if (dst->geom->label_font) {
